File: remove-element/removeElement.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function (nums, val) {
  let i = 0,
    iDest = 0,
    result = 0;

  while (i < nums.length) {
    if (nums[i] != val) {
      nums[iDest++] = nums[i];
      result++;
    }
    i++;
  }
  return result;
};

File: remove-element/RemoveElement.cs
public class Solution
{
    public int RemoveElement(int[] nums, int val)
    {
        var result = 0;
        var pointer = 0;
        foreach (var num in nums)
        {
            if (num != val)
            {
                nums[pointer++] = num;
                result++;
            }
        }
        return result;
    }
}